contrib/internet-urllib3.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+
+import urllib3
+import certifi
+
+def isInternetConnected(url="www.ietf.org"):
+  result = False
+  http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED', # Force certificate check.
+    ca_certs=certifi.where(),  # Path to the Certifi bundle.
+  )
+  try:
+    r = http.request('HEAD', 'https://' + url)
+    result = True
+  except Exception as e:  # urllib3.exceptions.SSLError
+    result = False
```

Comment:
The certificate is proof that we're really truly connected to the internet.  Without checking the certificate, your laptop can be connected to a "captive portal login page" which is returned by the gateway router, it spoofs all websites and returns a login page or a page "click here to accept terms and conditions to use the free wifi at xyz organization."